paasta_tools/cli/cmds/secret.py

```diff
@@ -48,75 +52,138 @@ def check_secret_name(secret_name_arg: str):
+    return arg
+
+
+def add_run_subparser(subparsers):
+    secret_parser_run = subparsers.add_parser(
+        "run",
+        help="runs a command with paasta secrets",
+        description=(
+            "Runs a command with the secret environment variables from "
+            "a given service instance. The command is run directly, not "
+            "in a Docker container. "
+            "Only the environment variables containing secrets are included. "
+            "No attempt at redacting secrets appearing in the output is made."
+        ),
+        conflict_handler="resolve",
     )
-    secret_parser.add_argument(
+    _add_common_args(secret_parser_run, allow_shared=False)
+
+    secret_parser_run.add_argument(
+        "-i",
+        "--instance",
+        help=(
+            "Instance of the service to retrieve secret environment variables "
+            "for, such as 'main' or 'canary'. Secrets will be selected and "
+            "mapped to environment variables based on the configs for the instance."
+        ),
+        required=True,
+    ).completer = lazy_choices_completer(list_instances)
+
+    secret_parser_run.add_argument(
         "-c",
         "--clusters",
-        help="A comma-separated list of clusters to create secrets for. "
-        "Note: this is translated to ecosystems because Vault is run "
-        "at an ecosystem level. As a result you can only have different "
-        "secrets per ecosystem. (it is not possible for example to encrypt "
-        "a different value for pnw-prod vs nova-prod. "
-        "Defaults to all clusters in which the service runs. "
-        "For example: --clusters pnw-prod,nova-prod ",
+        required=True,
+        type=_validate_single_cluster,
+        help=(
+            "The cluster to retrieve secrets for, e.g. norcal-devc. "
+            "A list of clusters is not supported for this command."
+        ),
     ).completer = lazy_choices_completer(list_clusters)
-    secret_parser.add_argument(
+
+    secret_parser_run.add_argument(
+        "cmd",
+        nargs="*",
+        default=["bash"],
+        help=(
+            "The command to run with the specified PaaSTA secrets. "
+            "If not given, starts an interactive bash shell."
+        ),
+    )
+
+    # see local_run.py
+    secret_parser_run.add_argument(
+        "--vault-auth-method",
+        help="Override how we auth with vault, defaults to token if not present",
+        type=str,
+        dest="vault_auth_method",
+        required=False,
+        default="token",  # token falls back to ldap if token file is unreadable
+        choices=["token", "ldap"],
+    )
+    secret_parser_run.add_argument(
+        "--vault-token-file",
+        help="Override vault token file, defaults to %(default)s",
+        type=str,
+        dest="vault_token_file",
+        required=False,
+        default="/var/spool/.paasta_vault_token",
+    )
```

Comment:
How does / will this work for "secrets for teams"?

Reply:
I haven't written in support for Secrets for Teams but that wouldn't be difficult to add in the future, there's already support in local_run.py and this would work the same way.

(These arguments wouldn't do anything in that case. In fact they should also be unnecessary even for vault secrets almost all the time, I considered setting `help=argparse.SUPPRESS` to make that clear.)